file-events/src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -66,8 +135,25 @@ WatchPoint::~WatchPoint() {
         SleepEx(0, true);
         close();
     } catch (const exception& ex) {
-        logToJava(LogLevel::WARNING, "Couldn't cancel watch point %s: %s", utf16ToUtf8String(path).c_str(), ex.what());
+        logToJava(LogLevel::WARNING, "Couldn't cancel watch point %s: %s", wideToUtf8String(registeredPath).c_str(), ex.what());
+    }
+}
+
+// Allocate maximum path length
+#define PATH_BUFFER_SIZE 32768
+
+wstring WatchPoint::getPath() {
+    vector<wchar_t> buffer;
+    buffer.reserve(PATH_BUFFER_SIZE);
+    DWORD pathLength = GetFinalPathNameByHandleW(
```

Comment:
I suppose there isn't a call which doesn't follow symlinks, since the file handle already points to the target of the symlink, right?

Reply:
I'm not sure about the reason, but yours sounds good. And indeed, I haven't found such a call. But as you can see from the changes to the tests, the behavior is now more like macOS/Linux, so I think it's fine. It doesn't matter for Gradle in any case, as we resolve the canonical file on the caller side when we register watchers.